File: src/test/java/io/github/numichi/reactive/logger/reactor/MDCSnapshotTest.java
package io.github.numichi.reactive.logger.reactor;

import io.github.numichi.reactive.logger.java.MDCSnapshot;
import org.junit.jupiter.api.Test;
import org.slf4j.MDC;

import java.util.HashMap;
import java.util.Map;
import java.util.UUID;

import static org.junit.jupiter.api.Assertions.assertEquals;

class MDCSnapshotTest {
    
    static String randomText() {
        return UUID.randomUUID().toString();
    }
    
    @Test
    void populateAndClear() {
        final Map<String, String> expected = new HashMap<>();
        expected.put(randomText(), randomText());
        expected.put(randomText(), randomText());
    
        assertEquals(new HashMap<>(), MDC.getCopyOfContextMap());
        
        try (final MDCSnapshot snapshot = MDCSnapshot.of(expected)) {
            assertEquals(MDC.getCopyOfContextMap(), expected);
        }
    
        assertEquals(new HashMap<>(), MDC.getCopyOfContextMap());
    }
    
    @Test
    void createEmptyInstance() {
        MDC.put(randomText(), randomText());
        
        try (final MDCSnapshot snapshot = MDCSnapshot.empty()) {
            assertEquals(new HashMap<>(), MDC.getCopyOfContextMap());
        }
    }
}

File: src/test/java/io/github/numichi/reactive/logger/reactor/MDCSnapshotTest.kt
package io.github.numichi.reactive.logger.reactor

import io.github.numichi.reactive.logger.reactor.MDCSnapshot.Companion.empty
import io.github.numichi.reactive.logger.reactor.MDCSnapshot.Companion.of
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test
import org.slf4j.MDC
import java.util.*

internal class MDCSnapshotTest {
    companion object {
        fun randomText(): String {
            return UUID.randomUUID().toString()
        }
    }

    @Test
    fun populateAndClear() {
        val expected: MutableMap<String, String> = mutableMapOf()
        expected[randomText()] = randomText()
        expected[randomText()] = randomText()
        assertEquals(mapOf<String, String>(), MDC.getCopyOfContextMap())

        of(expected).use {
            assertEquals(expected, MDC.getCopyOfContextMap())
        }

        assertEquals(mapOf<String, String>(), MDC.getCopyOfContextMap())
    }

    @Test
    fun createEmptyInstance() {
        MDC.put(randomText(), randomText())

        empty().use {
            assertEquals(mapOf<String, String>(), MDC.getCopyOfContextMap())
        }
    }
}